addons/website_event_booth/static/src/js/booth_register.js

```diff
@@ -196,37 +204,41 @@ publicWidget.registry.boothRegistration = publicWidget.Widget.extend({
         ev.preventDefault();
         ev.stopPropagation();
 
-        $(ev.currentTarget).addClass('disabled').attr('disabled', 'disabled');
-
-        const $form = this.$('#o_wbooth_contact_details_form');
-        if (this._isConfirmationFormValid($form)) {
-            const formData = new FormData($form[0]);
-            const response = await $.ajax({
-                url: `/event/${encodeURIComponent(this.$el.data('eventId'))}/booth/confirm`,
-                data: formData,
-                processData: false,
-                contentType: false,
-                type: 'POST',
-            });
-
-            const jsonResponse = response && JSON.parse(response);
+        ev.currentTarget.classList.add("disabled");
+        ev.currentTarget.disabled = true;
+
+        const formEl = this.el.querySelector("#o_wbooth_contact_details_form");
+        if (this._isConfirmationFormValid(formEl)) {
+            const formData = new FormData(formEl);
+            const response = await fetch(
```

Comment:
Can't we use `rpc(...)` instead of `fetch` ?

Reply:
@bso-odoo i dont think rpc will directly work with POST formData, as it is designed to send JSON payloads in the request body. Instead we can use [post](https://github.com/odoo/odoo/blob/17.0/addons/web/static/src/core/network/http_service.js#L18) method of http_service.js,  What is your view on this ?

Reply:
Oops, I did not notice this was the case.

Yes, `http_service`'s `post` sounds good.

Reply:
:sweat_smile:  Wonderful will make changes and ping you!

Reply:
@bso-odoo Also regarding [IMP]/[REF] we have simply converted the code and didn't introduced anything new to it, and According to (https://www.odoo.com/documentation/17.0/contributing/development/git_guidelines.html), [REF] is for a heavily rewritten feature. It seems that's not the case here. Thanks!